File: in_Python/0205 Isomorphic Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t): return False
        dict = {}
        for i in range(len(s)):
            a = s[i]
            b = t[i]
            if a not in dict and b not in dict.values():
                dict[a] = b
            if a in dict and dict[a] != b:
                return False
            if a not in dict and b in dict.values():
                return False
        return True

File: in_Java/0205 Isomorphic Strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character,Character> hm = new HashMap<>();
        for(int i = 0; i < s.length(); i++){
            char a = s.charAt(i);
            char b = t.charAt(i);
            if(!hm.containsKey(a)){
                if(hm.containsValue(b)) return false;
                else hm.put(a,b);
            }else{
                if(hm.get(a) != b) return false;
            }
        }
        return true;
    }
}